libyul/backends/evm/OptimizedEVMCodeTransform.cpp

```diff
@@ -351,7 +351,10 @@ void OptimizedEVMCodeTransform::createStackLayout(std::shared_ptr<DebugData cons
 				[&](JunkSlot const&)
 				{
 					// Note: this will always be popped, so we can push anything.
-					m_assembly.appendInstruction(evmasm::Instruction::CODESIZE);
+					if (m_assembly.evmVersion().hasPush0())
+						m_assembly.appendConstant(0);
+					else
+						m_assembly.appendInstruction(evmasm::Instruction::CODESIZE);
```

Comment:
I wonder if `PC` would have been nicer, given that is never used, and such is easier to detect at runtime (in case of a bug).

Reply:
Well, maybe, I won't change it here, though, in any case :-).